src/main/java/edu/hm/hafner/analysis/parser/MsBuildParser.java

```diff
@@ -25,7 +25,7 @@ public class MsBuildParser extends LookaheadParser {
             + ANT_TASK + "(?:(?:\\s*(?:\\d+|\\d+:\\d+)>)?(?:(?:(?:(?<a4>.*?)\\((?<a5>\\d*)(?:,(?<a6>\\d+))?[a-zA-Z]*?\\)|.*LINK)\\s*:|"
             + "(?<a7>.*):)\\s*(?<a8>[A-z-_]*\\s(?:[Nn]ote|[Ii]nfo|[Ww]arning|(?:fatal\\s*)?[Ee]rror))[^A-Za-z0-9]\\s*:?\\s*(?<a9>[A-Za-z0-9\\-_]+)?"
             + "\\s*:\\s(?:\\s*(?<a10>[A-Za-z0-9.]+)\\s*:)?\\s*(?<a11>.*?)(?: \\[(?<a12>[^\\]]*)[/\\\\][^\\]\\\\]+\\])?"
-            + "|(?<a13>.*)\\s*(?<severity>(error|warning))\\s*:\\s*(?<a15>LNK[0-9]+):\\s*(?<a16>.*)))$";
+            + "|(?<a13>.*)\\s*:.*(?<severity>(error|warning))\\s*:\\s*(?<a15>LNK[0-9]+):\\s*(?<a16>.*)))$";
```

Comment:
I tried to add space after `(error|warning)` because the test file has space. 

To remove `:` from being part of error and warning I added it after space. 

After lot of research this what I could come up with. 

The test still fails.

Reply:
Since you are not using the groups it cannot work. As mentioned above your groups need to be identifiers and these identifiers need to be used to get the matches. Did you try in the debugger?